src/scores/probability/rev_impl.py

```diff
@@ -0,0 +1,812 @@
+        Society, 126(563), 649-667. https://doi.org/10.1002/qj.49712656313
+
+    Examples:
+        Calculate REV from pre-computed rates:
+
+        >>> import numpy as np
+        >>> import xarray as xr
+        >>> from scores.probability import relative_economic_value_from_rates
+        >>>
+        >>> # Pre-computed detection rates
+        >>> pod = xr.DataArray([0.8, 0.6, 0.4], dims=['threshold'])
+        >>> pofd = xr.DataArray([0.2, 0.1, 0.05], dims=['threshold'])
+        >>> climatology = xr.DataArray(0.3)  # 30% base rate
+        >>>
+        >>> # Calculate REV at multiple cost-loss ratios
+        >>> cost_loss_ratios = [0.1, 0.3, 0.5, 0.7, 0.9]
+        >>> rev = relative_economic_value_from_rates(
+        ...     pod, pofd, climatology, cost_loss_ratios
+        ... )
+        >>> rev.dims
+        ('cost_loss_ratio', 'threshold')
+
+        Calculate REV values:
+
+        >>> # Perfect forecast scenario
+        >>> pod_perfect = xr.DataArray(1.0)
+        >>> pofd_perfect = xr.DataArray(0.0)
+        >>> climatology = xr.DataArray(0.5)
+        >>>
+        >>> rev = relative_economic_value_from_rates(
+        ...     pod_perfect, pofd_perfect, climatology,
+        ...     cost_loss_ratios=[0.5],
+        ... )
+        >>> rev.values  # Returns finite value close to 1
+        array([1.])
+
+    See Also:
+        - :py:func:`scores.probability.relative_economic_value`
+        - :py:func:`scores.categorical.probability_of_detection`
+        - :py:func:`scores.categorical.probability_of_false_detection`
+    """
+
+    # From Richardson, D.S., QJR Meteorol Soc 2000, 126, pp 649-667
+    # Hit rate (probability of detection) equation 4
+    # False alarm rate (probability of false detection) equation 5
+
+    try:
+        check_monotonic_array(cost_loss_ratios)
+    except Exception as ex:
+        raise type(ex)("for cost_loss_ratios, " + str(ex))
+
+    if cost_loss_dim in (set(pod.dims) | set(pofd.dims) | set(climatology.dims)):
+        raise ValueError(f"dimension '{cost_loss_dim}' must not be in input data")
+
+    # Handle Dataset inputs by applying to each variable
+    if isinstance(pod, xr.Dataset):
+        result_dict = {}
+        for var in pod.data_vars:
+            result_dict[var] = relative_economic_value_from_rates(
+                pod[var],
+                pofd[var] if isinstance(pofd, xr.Dataset) else pofd,
+                climatology[var] if isinstance(climatology, xr.Dataset) else climatology,
+                cost_loss_ratios,
+                cost_loss_dim=cost_loss_dim,
+            )
+        return xr.Dataset(result_dict)
+
+    alphas = xr.DataArray(
+        cost_loss_ratios,
+        dims=[cost_loss_dim],
+        coords={cost_loss_dim: cost_loss_ratios},
+    )
+
+    obar, alphas = xr.broadcast(climatology, alphas)
+    climatological_term = obar.copy()
+    climatological_term.values = np.minimum(obar.values, alphas.values)
```

Comment:
```suggestion
climatological_term = xr.where(obar < alphas, obar, alphas)
```
You can simplify this to avoid the `.copy()`

Reply:
Accepted as suggested